src/main/java/ch/uzh/ifi/hase/soprafs22/lobby/Lobby.java

```diff
@@ -132,6 +128,43 @@ public void assignNewHost() {
         }
     }
 
+    @Override
+    public void reduceLobbyCapacity() {
+        Iterator <IPlayer> it = iterator();
+        while(it.hasNext()){
+            if(it.next() != host && lobbyCapacity < getNumberOfPlayers()){
+                it.remove();
+            }
+        }
+    }
+
+    @Override
+    public void setAllPlayersNotReady(){
+        for (IPlayer player : playerMap.values()){
+            player.setReady(false);
+        }
+    }
+
+    @Override
+    public void balanceTeams(){
+        Team t = null;
+        // we alternate teams between consecutive players
+        for (IPlayer player : playerMap.values()){
+            if (t == null){
+                player.setTeam(Team.values()[0]);
```

Comment:
you could use the enum values couldn't you?
Team.RED and Team.BLUE instead of Team.values()[0/1]